lib/jira/install.js

```diff
@@ -1,28 +1,43 @@
 const { Installation } = require('../models')
 const getAxiosInstance = require('../jira/client/axios')
+const jwt = require('./util/jwt')
 
 module.exports = async (req, res) => {
   req.log('Received installation payload.')
   const { baseUrl: host, clientKey, sharedSecret } = req.body
-  res.sendStatus(204)
+  let successfulRes = false;
+
+  try {
+    const hostData = await Installation.getForClientKey(clientKey)
+    if (jwt.hasValidJwt(hostData.sharedSecret, host, req, res)) {
+      res.sendStatus(204)
+      successfulRes = true
```

Comment:
This should be `false`, right? If the installation already exists and is valid, we don't want to store a new one. We could even `return` here.

Reply:
Unfortunately some information can change in the installation flow.
If the app is uninstalled from a site, then reinstalled, the JWT will be signed with the old sharedSecret but a new sharedSecret will be supplied.

Also there is a chance that the baseUrl can change (site rename) and a new installation is sent to update the app